File: app/src/main/java/maderski/chargingindicator/Receivers/BootCompletedReceiver.java
package maderski.chargingindicator.Receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import maderski.chargingindicator.Services.CIService;

/**
 * Created by Jason on 3/10/17.
 */

public class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        context.startService(new Intent(context, CIService.class));
    }
}

File: app/src/main/java/maderski/chargingindicator/receivers/BootCompletedReceiver.kt
package maderski.chargingindicator.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.work.Constraints
import androidx.work.OneTimeWorkRequestBuilder
import androidx.work.WorkManager
import maderski.chargingindicator.workers.OnBootWorker

class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if (context != null && intent != null) {
            val action = intent.action
            if (action == Intent.ACTION_BOOT_COMPLETED) {
                // Create and add work request to work manager
                val workOnBootRequest = OneTimeWorkRequestBuilder<OnBootWorker>()
                        .addTag(OnBootWorker.TAG)
                        .build()
                WorkManager.getInstance().enqueue(workOnBootRequest)
            }
        }
    }
}